centreon/src/Core/TimePeriod/Infrastructure/API/AddTimePeriod/AddTimePeriodController.php

```diff
@@ -25,97 +25,49 @@
+    ): Response {
         $this->denyAccessUnlessGrantedForApiConfiguration();
-        try {
-            /**
-             * @var array{
-             *     name: string,
-             *     alias: string,
-             *     days: array<array{
-             *         day: integer,
-             *         time_range: string
-             *     }>,
-             *     templates: int[],
-             *     exceptions: array<array{
-             *         day_range: string,
-             *         time_range: string
-             *     }>
-             * } $dataSent
-             */
-            $dataSent = $this->validateAndRetrieveDataSent($request, __DIR__ . '/AddTimePeriodSchema.json');
-            $dtoRequest = $this->createDtoRequest($dataSent);
-            $useCase($dtoRequest, $presenter);
-        } catch (\InvalidArgumentException $ex) {
-            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
-            $presenter->setResponseStatus(new InvalidArgumentResponse($ex));
-        }
+        $useCase($this->createDto($input), $presenter);
 
         return $presenter->show();
     }
 
-    /**
-     * @param array{
-     *     name: string,
-     *     alias: string,
-     *     days: array<array{
-     *         day: integer,
-     *         time_range: string
-     *     }>,
-     *     templates: int[],
-     *     exceptions: array<array{
-     *         day_range: string,
-     *         time_range: string
-     *     }>
-     * } $dataSent
-     *
-     * @return AddTimePeriodRequest
-     */
-    private function createDtoRequest(array $dataSent): AddTimePeriodRequest
+    private function createDto(AddTimePeriodRequest $data): AddTimePeriodDto
     {
-        $dto = new AddTimePeriodRequest();
-        $dto->name = $dataSent['name'];
-        $dto->alias = $dataSent['alias'];
-        $dto->days = array_map(
-            fn (array $day): array => ['day' => $day['day'], 'time_range' => $day['time_range']],
-            $dataSent['days']
-        );
-        $dto->templates = $dataSent['templates'];
-        $dto->exceptions = array_map(
-            fn (array $exception): array => [
-                'day_range' => $exception['day_range'],
-                'time_range' => $exception['time_range'],
-            ],
-            $dataSent['exceptions']
+        return new AddTimePeriodDto(
+            is_string($data->name) ? $data->name : '',
+            is_string($data->alias) ? $data->alias : '',
+            array_map(
+                fn (array $day): array => ['day' => $day['day'], 'time_range' => $day['time_range']],
+                $data->days ?? []
+            ),
+            $data->templates ?? [],
+            array_map(
+                fn (array $exception): array => [
+                    'day_range' => $exception['day_range'],
+                    'time_range' => $exception['time_range'],
+                ],
+                $data->exceptions ?? []
+            ),
         );
```

Comment:
I probably would've implemented a `public function toDto(): AddTimeRequestDto` method in AddTimePeriodRequest to avoid polluting the controller with Dto instanciation.